### PR TITLE
ci(jenkins): enable pull requests using the upstream repo

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- It will enable to use the integrations tests either from pull requests which are coming from a forked repo or from this one.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ pipeline {
       agent { label 'master || immutable' }
       options { skipDefaultCheckout() }
       steps {
+        sh 'env | sort'
```
Comment (v1v) [inline Jenkinsfile]: This is only for testing purposes as I wanna be sure what's the env variable at runtime. I'll remove it later on

Comment (v1v) [reply Jenkinsfile]:
```suggestion
```

**Comment by watson** [comment]
@v1v the regular Jenkins tests doesn't currently run because if the commit message linting error (see #1157)

**Comment by watson** [comment]
@v1v just out of curiosity, what was the reason why the two Jenkins failures wasn't a problem for merging?

**Comment by v1v** [comment]
There were a couple of infra errors during the TAV stage
- https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp%2FPR-1170/detail/PR-1170/4/pipeline/885#log-656

And also a error in the build for the branch rather than for the PR.  `continuous-integration/jenkins/branch` is the github check which points out to the branch. while `
continuous-integration/jenkins/pr-merge ` is the one which points out to the PR.

So I manually triggered another build which passed:
- https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/PR-1170/5/pipeline

Unfortunately, the GitHub checks are not updated unless the same stages have been triggered in the following builds. In this case `#5` launched a subset of stages:
- no docs stage
- no tav stage

Further details [here](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-1170/5/parameters/)

Please let me know if I did miss anything

**Pull request merged**